### PR TITLE
Move Identity tests from SQL Server to SQLite for better reliability

### PR DESCRIPTION
**Pull request by ajcvickers**

### CODE REVIEW

File: src/Identity/EntityFrameworkCore/test/EF.Test/DbUtil.cs [hunk 1]
```diff
@@ -1,76 +1,71 @@
...
 
-        public static bool VerifyMaxLength(SqlConnection conn, string table, int maxLength, params string[] columns)
+        public static bool VerifyMaxLength(DbContext context, string table, int maxLength, params string[] columns)
         {
             var count = 0;
-            using (
-                var command =
-                    new SqlCommand("SELECT COLUMN_NAME, CHARACTER_MAXIMUM_LENGTH FROM INFORMATION_SCHEMA.COLUMNS where TABLE_NAME=@Table", conn))
+
+            foreach (var property in context.Model.GetEntityTypes().Single(e => e.Relational().TableName == table).GetProperties())
```
Comment (HaoK) [inline src/Identity/EntityFrameworkCore/test/EF.Test/DbUtil.cs]: this looks much cleaner now :)

File: src/Identity/EntityFrameworkCore/test/EF.Test/SqlStoreTestBase.cs [hunk 2]
```diff
@@ -58,11 +59,6 @@ protected override void SetupIdentityServices(IServiceCollection services, objec
             SetupAddIdentity(services);
         }
 
-        protected override bool ShouldSkipDbTests()
```
Comment (HaoK) [inline src/Identity/EntityFrameworkCore/test/EF.Test/SqlStoreTestBase.cs]: Nice so we also get cross plat testing finally

**Comment by ajcvickers** [comment]
@HaoK Any idea why I am seeing:
```
/home/vsts/work/1/s/src/Security/Authentication/Negotiate/samples/NegotiateAuthSample/Startup.cs(18,22): error CS1061: 'IServiceCollection' does not contain a definition for 'AddAuthorization' and no accessible extension method 'AddAuthorization' accepting a first argument of type 'IServiceCollection' could be found (are you missing a using directive or an assembly reference?) [/home/vsts/work/1/s/src/Security/Authentication/Negotiate/samples/NegotiateAuthSample/NegotiateAuthSample.csproj]
/home/vsts/work/1/s/src/Security/Authentication/Negotiate/samples/NegotiateAuthSample/Startup.cs(30,17): error CS1061: 'IApplicationBuilder' does not contain a definition for 'UseAuthorization' and no accessible extension method 'UseAuthorization' accepting a first argument of type 'IApplicationBuilder' could be found (are you missing a using directive or an assembly reference?) [/home/vsts/work/1/s/src/Security/Authentication/Negotiate/samples/NegotiateAuthSample/NegotiateAuthSample.csproj]
```

**Comment by ajcvickers** [comment]
@dougbu @natemcmaster Or is it that the AspNet build is broken?

**Comment by HaoK** [comment]
Probably broken, @Tratcher just merged Negotiate handler in

**Comment by Tratcher** [comment]
@HaoK you just moved the AuthZ components, no? My PR hadn't rebuilt since. I'll update it.

**Comment by Tratcher** [comment]
https://github.com/aspnet/AspNetCore/pull/10453

File: src/Identity/EntityFrameworkCore/test/EF.Test/Microsoft.AspNetCore.Identity.EntityFrameworkCore.Test.csproj [hunk 3]
```diff
@@ -20,8 +20,7 @@
     <Reference Include="Microsoft.AspNetCore.Hosting" />
     <Reference Include="Microsoft.AspNetCore.Http" />
     <Reference Include="Microsoft.AspNetCore.TestHost" />
-    <Reference Include="Microsoft.EntityFrameworkCore.InMemory" />
-    <Reference Include="Microsoft.EntityFrameworkCore.SqlServer" />
```
Comment (natemcmaster) [inline src/Identity/EntityFrameworkCore/test/EF.Test/Microsoft.AspNetCore.Identity.EntityFrameworkCore.Test.csproj]: 👍 This means you can also remove the line above that says "TestDependsOnMssql".

**Review by natemcmaster — APPROVED** [review]
LGTM. Thanks!

**Comment by ajcvickers** [comment]
@natemcmaster I may be missing something here, but it looks like the PR checks are not running.

**Comment by natemcmaster** [comment]
https://www.githubstatus.com/incidents/nf4gcjtcsqdb

File: src/Mvc/Mvc.Testing/ref/Microsoft.AspNetCore.Mvc.Testing.netcoreapp3.0.cs [hunk 4]
```diff
@@ -36,12 +36,6 @@ public partial class WebApplicationFactory<TEntryPoint> : System.IDisposable whe
         protected virtual Microsoft.Extensions.Hosting.IHost CreateHost(Microsoft.Extensions.Hosting.IHostBuilder builder) { throw null; }
         protected virtual Microsoft.Extensions.Hosting.IHostBuilder CreateHostBuilder() { throw null; }
         protected virtual Microsoft.AspNetCore.TestHost.TestServer CreateServer(Microsoft.AspNetCore.Hosting.IWebHostBuilder builder) { throw null; }
-        protected virtual Microsoft.AspNetCore.Hosting.IWebHostBuilder CreateWebHostBuilder() { throw null; }
```
Comment (natemcmaster) [inline src/Mvc/Mvc.Testing/ref/Microsoft.AspNetCore.Mvc.Testing.netcoreapp3.0.cs]: This is causing compiler errors. Looks like a bad rebase.

Comment (natemcmaster) [reply src/Mvc/Mvc.Testing/ref/Microsoft.AspNetCore.Mvc.Testing.netcoreapp3.0.cs]: Arthur is OOF right now. I'm going to update this PR and see if we can get it merged today.

**Comment by dougbu** [comment]
First validation attempt (https://dev.azure.com/dnceng/public/_build/results?buildId=199037) failed due to compiler race condition (NRE in an analyzer). Retried&hellip;

**Comment by natemcmaster** [comment]
Just merged the fix to the compiler race in master. I'm comfortable merging this now.

**Comment by natemcmaster** [comment]
Last test failure should be resolved by #10483. Ignoring that test failure and merging this.

**Comment by ajcvickers** [comment]
Thanks @natemcmaster and @dougbu for getting this in!

**Pull request merged**